### PR TITLE
Changelog November 2020 update

### PR DESCRIPTION
**Pull request by kevingranade**
#### Summary
None

#### Purpose of change
Update the changelog for the month of November

#### Describe the solution
tools/generate_changelog.py -f --by-date changelog_11  -t ../repo_token -e 2020-11-01 2020-10-01
tac changelog_11 > changelog_11_rev
tools/merge_summaries.sh changelog_11_rev data/changelog.txt
Then copyedit the results.

#### Additional context
Typos happen, and some of the things I prune might be bigger than I remember or part of some larger set of changes that should go in the changelog, please take a look at not just the final diff, but also the copyedit commit to check for missed entries.

This includes the previous PR since the new updates are interleaved with the old ones.

Also includes a minor but critical fix to the changelog generator script.

### CODE REVIEW

File: data/changelog.txt [hunk 1]
```diff
@@ -71,6 +78,15 @@ Display missing and obsolete mods for current world.
 Require grenades and other explosives to be wielded before activation.
 Add bigger mini-maps to all sidebar layouts.
 Add keycode support to uilist.
+Allow disabling robots by 'e'xamining them.
+Warn if friendly creature or non-hostile NPC enters line of fire.
+Replaces the CAL/kg row of the (E)at menu with a more meaningful Satiety row.
```
Comment (actual-nh) [inline data/changelog.txt:83]:
```suggestion
Replaces the CAL/kg column of the (E)at menu with a more meaningful Satiety column.
```

File: data/changelog.txt [hunk 2]
```diff
@@ -71,6 +78,15 @@ Display missing and obsolete mods for current world.
...
+Add a Mod to use MSX Characters and monster with the Ultica tileset
```
Comment (actual-nh) [inline data/changelog.txt:84]:
```suggestion
Adds a Mod to use MSX characters and monsters with the Ultica tileset.
```
Still is not quite clear to me - will take a look at original PR.

File: data/changelog.txt [hunk 3]
```diff
@@ -71,6 +78,15 @@ Display missing and obsolete mods for current world.
...
+Show missing proficiency failure rates for crafting.
+Display contained items like 'water (plastic bottle)' instead of 'plastic bottle of water'.
+Persist preferred aiming mode.
+Make Chibi Ultica a Stand-alone tileset
+Bring back MSX+
```
Comment (actual-nh) [inline data/changelog.txt:89]:
```suggestion
Makes Chibi Ultica a stand-alone tileset.
Brings back MSX+.
```

File: data/changelog.txt [hunk 4]
```diff
@@ -80,6 +96,26 @@ Overhaul Blazemod and split into seperate mods.
...
+Adds location variants with dinosaurs.
+Aftershock: Megacity oriented overmap generation adjustments
+Aftershock: New Melee oriented bionics
+Adds Acrocanthosaurus, dino baby overhaul.
+Add New Black Dragon Mutation Line.
+Adjust fog frequency in black dragon lair, minor map adjustments.
+Adds monsters 'Scutellosaurus', 'Maiasaura', and 'Coelophysis'.
+[Magiclysm] Attunements - Special abilities gained by mastering two schools of magic.
+Add monsters 'Pachyrhinosaurus', 'Pentaceratops', and 'Torosaurus'
+Organizes dinos into three main factions with defined behavior
```
Comment (actual-nh) [inline data/changelog.txt:108]:
```suggestion
Adds monsters 'Pachyrhinosaurus', 'Pentaceratops', and 'Torosaurus'.
Organizes dinos into three main factions with defined inter-faction behavior.
```

File: data/changelog.txt [hunk 5]
```diff
@@ -80,6 +96,26 @@ Overhaul Blazemod and split into seperate mods.
 Polymorph spells to transform your foes.
 Spells using components system.
 Aftershock: Add autodoc tiers and augmentation clinics.
+Adds location variants with dinosaurs.
+Aftershock: Megacity oriented overmap generation adjustments
+Aftershock: New Melee oriented bionics
+Adds Acrocanthosaurus, dino baby overhaul.
+Add New Black Dragon Mutation Line.
+Adjust fog frequency in black dragon lair, minor map adjustments.
```
Comment (actual-nh) [inline data/changelog.txt:104]:
```suggestion
Adds New Black Dragon Mutation Line.
Adjusts fog frequency in black dragon lair plus other (minor) map adjustments.
```

File: data/changelog.txt [hunk 6]
```diff
@@ -80,6 +96,26 @@ Overhaul Blazemod and split into seperate mods.
 Polymorph spells to transform your foes.
 Spells using components system.
 Aftershock: Add autodoc tiers and augmentation clinics.
+Adds location variants with dinosaurs.
+Aftershock: Megacity oriented overmap generation adjustments
+Aftershock: New Melee oriented bionics
```
Comment (actual-nh) [inline data/changelog.txt:101]:
```suggestion
Aftershock: Adds new Melee oriented bionics.
```

File: data/changelog.txt [hunk 7]
```diff
@@ -80,6 +96,26 @@ Overhaul Blazemod and split into seperate mods.
...
+Rebalance Jurassic spawns and monsters for realism, fix predator in-fighting.
+Cap monsters at 1000 kg so butchering kind of works.
+[Magiclysm] implement summonable motorcycle
+[Magiclysm] Adds a new house with technomancer themed loot.
+Aftershock: Aftershock: Replace Vanilla autodocs with Aftershock Average Autodocs
+Adds monsters 'Brontosaurus', 'Alamosaurus', and 'Brachiosaurus'
+[magiclysm] rings of wizardry, mojocycle tweaks
+[magiclysm] add bullet armor to magic rings and bracers of protection
+Convert 'dinoexhibit' to use templates, add roof, add baby dino variant
+Ankylosaurs Abound: Dyoplosaurus Nodosaurus Edmontonia
```
Comment (actual-nh) [inline data/changelog.txt:118]:
```suggestion
[Magiclysm] Adds a new house with technomancer-themed loot.
Aftershock: Replaces Vanilla autodocs with Aftershock Average Autodocs.
DinoMod: Adds monsters 'Brontosaurus', 'Alamosaurus', and 'Brachiosaurus'.
[Magiclysm] Rings of wizardry and mojocycle tweaks
[Magiclysm] Adds bullet armor to magic rings and bracers of protection.
DinoMod: Converts 'dinoexhibit' to use templates, adds a roof, and adds a baby dino variant.
Ankylosaurs Abound: Dyoplosaurus Nodosaurus Edmontonia
```

File: data/changelog.txt [hunk 8]
```diff
@@ -90,6 +126,16 @@ More complex satiety calculations.
 NPCs to use ranged weapons against all enemies, and automatic fire against dangerous enemies.
 Antifungal drugs work over time.
 Don't damage hands when smashing but use your most armored part instead.
+Guarantee at least one transversable path in the collapsed tower basement.
+Make Solar Sensitivity mutation respect clothing coverage, precious.
+Unlock NPC recruiting when you build the faction camp radio console.
+Clean_water recipe with purification tablets or charcoal water purifier.
```
Comment (actual-nh) [inline data/changelog.txt:132]:
```suggestion
Fixed clean_water recipes with purification tablets and charcoal water purifier.
```
Or did it add them?

File: data/changelog.txt [hunk 9]
```diff
@@ -90,6 +126,16 @@ More complex satiety calculations.
...
+Bring reloading components close to reality, make brass spawn in packs, make gunpowder units 100 mg
```
Comment (actual-nh) [inline data/changelog.txt:133]:
```suggestion
Brings reloading components closer to reality, make brass casings spawn in packs, and makes gunpowder units 100 mg.
```
"Casings" may need double-checking.

File: data/changelog.txt [hunk 10]
```diff
@@ -138,12 +184,31 @@ Vehicles: make shock absorbers actually absorb shock damage.
 Fix NPC dying immediately on accepting inhaler mission
 Fix crash when attempting to butcher a corpse.
 Fix gear getting damaged when weather has no damaging effects.
+Improve invalidation of max_populated_zlev cache, fixes lighting in some teleportation scenarios.
+Prevent dodging or blocking while insensible.
+Fix apparent light asymmetry when sun shines through a single-tile hole in the roof.
```
Comment (actual-nh) [inline data/changelog.txt:189]:
```suggestion
Fixes apparent light asymmetry when the sun shines through a single-tile hole in the roof.
```

File: data/changelog.txt [hunk 11]
```diff
@@ -138,12 +184,31 @@ Vehicles: make shock absorbers actually absorb shock damage.
...
+Made installation time mandatory for gunmods.
+Fix #44421 tents and vehicles are not blocking sunlight.
```
Comment (actual-nh) [inline data/changelog.txt:191]:
```suggestion
Makes tents and vehicles block sunlight.
```

File: data/changelog.txt [hunk 12]
```diff
@@ -138,12 +184,31 @@ Vehicles: make shock absorbers actually absorb shock damage.
...
+Fix scent map saved with invalid utf8 sequence.
+Prevent monsters from bashing obstacles based on tracking sounds.
```
Comment (actual-nh) [inline data/changelog.txt:193]: This is a bit unclear. "based on" is meaning "only due to", or what?

File: data/changelog.txt [hunk 13]
```diff
@@ -138,12 +184,31 @@ Vehicles: make shock absorbers actually absorb shock damage.
...
+Picking up an item stack fills up your pockets if it doesn't fit into one pocket.
+Wielding combines item stacks if possible.
+Robots can no longer be disabled through melee hits.
+Fix auto consume zones in vehicles.
```
Comment (actual-nh) [inline data/changelog.txt:197]:
```suggestion
Fixes auto-consume zones in vehicles.
```

File: data/changelog.txt [hunk 14]
```diff
@@ -138,12 +184,31 @@ Vehicles: make shock absorbers actually absorb shock damage.
...
 Add option to enable batching when it is available in SDL.
 Skip unnecessary weather calculations for root cellar.
 Cache drop_invalid_inventory and weight_carried.
-Check for safemode emptiness before calling monster check.
+Cache clairvoyance field.
+Performance improvements related to map::process_fields.
+Overhaul shadowcasting to be iterative instead of recursive.
+Use plf linked list impl for shadowcasting.
+Improve string_id comparison performance, add unit test and benchmarks.
+Improve rendering speed by optimizing find_tile_looks_like method (FPS improved ≈15%)
```
Comment (actual-nh) [inline data/changelog.txt:208]:
```suggestion
Improves rendering speed by optimizing find_tile_looks_like method (FPS improved ~15%)
```
Make pure ASCII.

File: data/changelog.txt [hunk 15]
```diff
@@ -138,12 +184,31 @@ Vehicles: make shock absorbers actually absorb shock damage.
...
+Greatly reduce overhead of processing idle vehicles.
+Optimize Creature::has_effect (performance improved ≈9.5%)
+Add a specialization for static/interned string_ids  (performance improved ≈30%))
```
Comment (actual-nh) [inline data/changelog.txt:211]:
```suggestion
Optimizes Creature::has_effect (performance improved ~9.5%)
Adds a specialization for static/interned string_ids  (performance improved ~30%))
```

File: data/changelog.txt [hunk 16]
```diff
@@ -201,7 +266,14 @@ Add error-checking of recipe (sub)categories.
 Unblacklist most remaining blacklisted mods for automated consistency checking.
 Report log messages for failed tests.
 Reinstate blueprint autocalc.
-Add a compiler error problem matcher.
+Refactored vision_test. Introduced map_test_case framework to build similar tests easier.
+Add tests for better_pocket and best_pocket.
+Rename skills in preparation of some changes; merge speech & barter.
```
Comment (actual-nh) [inline data/changelog.txt:271]:
```suggestion
Renames skills in preparation for some future improvements; merges speech & barter.
```

File: data/changelog.txt [hunk 17]
```diff
@@ -201,7 +266,14 @@ Add error-checking of recipe (sub)categories.
...
+Merge trapping and lockpicking skills into a new 'devices' skill, expand scope somewhat.
```
Comment (actual-nh) [inline data/changelog.txt:272]:
```suggestion
Merges trapping and lockpicking skills into a new 'devices' skill, and expands its scope somewhat.
```

File: data/changelog.txt [hunk 18]
```diff
@@ -217,6 +289,10 @@ Improve use of precompiled headers.
 compose.py: allow default sized tilesheets to specify offsets.
 Enable compiling an Android build under Windows.
 Tools: add a Python script to reformat vehicles to use parts arrays.
+Suppress Homebrew update for faster OSX builds.
+Add problem matchers for json errors, compiler errors, DebugLog emission and Catch2 errors.
+Add check enforcing almost-never-auto convention.
+Add a GitHub workflow running flake8 on the Python scripts.
```
Comment (actual-nh) [inline data/changelog.txt:295]:
```suggestion
Adds a GitHub workflow running flake8 to check the Python scripts.
```

File: data/changelog.txt [hunk 19]
```diff
@@ -217,6 +289,10 @@ Improve use of precompiled headers.
 compose.py: allow default sized tilesheets to specify offsets.
 Enable compiling an Android build under Windows.
 Tools: add a Python script to reformat vehicles to use parts arrays.
+Suppress Homebrew update for faster OSX builds.
+Add problem matchers for json errors, compiler errors, DebugLog emission and Catch2 errors.
```
Comment (actual-nh) [inline data/changelog.txt:293]:
```suggestion
Adds problem matchers for json errors, compiler errors, DebugLog emissions, and Catch2 errors.
```

File: data/changelog.txt [hunk 20]
```diff
@@ -217,6 +289,10 @@ Improve use of precompiled headers.
...
+Add check enforcing almost-never-auto convention.
+Add a GitHub workflow running flake8 on the Python scripts.
 
 ## I18N and A11Y:
 Fix translation of martial art initate message and computer translation.
```
Comment (actual-nh) [inline data/changelog.txt:298]:
```suggestion
Fix translation of martial art initiate message and computer translation.
```

**Review by actual-nh — COMMENTED** [review]
Should initial verbs be singular or plural? I feel that they should be plural - "[the update] adds", for instance. I can either go back and remove the changes I've already done on that, or go through the other lines and change them.

File: data/changelog.txt (hunk 12 above)
Comment (kevingranade) [reply data/changelog.txt:193]: Only due to, i.e. they won't bash things unless they see or smell something, I agree it needs clarification, it's a pretty big change.
```suggestion
Monsters no longer bash obstacles based on sound. This prevents groups of monsters from getting stuck in a loop of bashing based on the sound of other monsters bashing.
```

**Pull request merged**